backend/copr_backend/pulp.py

```diff
@@ -363,6 +366,59 @@ def create_content(self, path, labels):
             package = self.send("POST", uri, data=data, files=files)
         return package
 
+    def create_content_chunked(self, path, labels):
+        # TODO use labels
+
+        # TODO
+        uri = "/api/v3/uploads/"
+        data = {
+            "size": os.path.getsize(path),
+        }
+        response = self.send("POST", uri, data)
+
+        upload_href = response.json()["pulp_href"]
+        upload_url = self.config["base_url"] + upload_href
+
+        # TODO
+        size = 10000
+        sha256 = hashlib.sha256()
+        chunks = enumerate(self._read_chunks(path, size=size))
+        for i, chunk in chunks:
+            start = i * size
+            end = start + len(chunk) - 1
+            headers = {
+                "Content-Range": "bytes {0}-{1}/*".format(start, end),
+            }
+            files = {
+                "file": ("chunk{0}".format(i), io.BytesIO(chunk), "application/octet-stream"),
+            }
+            response = self.send("PUT", upload_url, files=files, headers=headers)
+            sha256.update(chunk)
+
+
+        # TODO
+        commit_url = self.config["base_url"] + upload_href + "commit/"
+        data = {
+            "sha256": sha256.hexdigest(),
```

Comment:
Now I’m a bit confused. Is PULP responsible for verifying the SHA-256? If so, I think we shouldn’t calculate the SHA-256 "per chunk", but only for the original complete file, and let PULP confirm we haven't made any mistake.

Reply:
Updated with:

```python
with open(path, "rb") as fp:
    sha256 = hashlib.file_digest(fp, "sha256")
```